cli/azd/pkg/tools/azcli.go

```diff
@@ -21,7 +21,8 @@ import (
 )
 
 var (
-	ErrAzCliNotLoggedIn          = errors.New("cli is not logged in")
+	ErrAzCliNotLoggedIn          = errors.New("cli is not logged in. Try running \"azd login\" to fix")
+	ErrAzCliRefreshTokenExpired  = errors.New("refresh token has expired. Try running \"azd login\" to fix")
```

Comment:
I usually don't like to create errors before the error has happened.
@ellismg is this a common pattern in Golang?  

Coming from C++, it makes me feel like we are filling the Heap with objects that we might not require.

Instead of this pattern, I would recommend using a function that returns the error (whenever we need to re-use the error) or just creating the error at the moment if has happened.

Reply:
>@ellismg is this a common pattern in Golang?

Yes - it is common.  Doing things this way allows us to use `errors.Is` to test if a specific type of error has been returned (for example, see the built in `io` package which follows this same pattern: https://pkg.go.dev/io#pkg-variables)

If we were to instead create the error when we need it, it would mean each time we want to return an error we call `errors.New` which creates a new object and there would be no way to test to see if it was a well known error.

This works when the error message is "stable".  When you want to add instance specific context and still allow callers to detect the error case programmatically, the practice is to create a new error type, export it and then return instances of it, and callers can detect this case using `errors.As`, but the ergonomics there are worse.  So in cases like this, I think this is the prefered pattern.